GH-139922: update 3.15 whatsnew: Windows 64-bit binaries now use the tail-calling interpreter (#146391)
python/cpython: Doc/whatsnew/3.15.rst

.. _whatsnew315-windows-tail-calling-interpreter:

* 64-bit builds using Visual Studio 2026 (MSVC 18) may now use the new
  :ref:`tail-calling interpreter <whatsnew314-tail-call-interpreter>`.
  Results on Visual Studio 18.1.1 report between
  `15-20% <https://github.com/faster-cpython/ideas/blob/main/results/5800X-msvc.pgo2-vs-msvc.pgo.tc.svg>`__
  speedup on the geometric mean of pyperformance on Windows x86-64 over
  the switch-case interpreter on an AMD Ryzen 7 5800X. We have
  observed speedups ranging from 14% for large pure-Python libraries
  to 40% for long-running small pure-Python scripts on Windows.
  This was made possible by a new feature introduced in MSVC 18,
  which the official Windows 64-bit binaries on python.org__ now use.
  (Contributed by Chris Eibl, Ken Jin, and Brandt Bucher in :gh:`143068`.

>   Special thanks to the MSVC team including Hulon Jenkins.)

Reviewer: Please change the line to:

`Special thanks to Steve Dower, and the MSVC team including Hulon Jenkins.`

Author: Yes of course! My bad, thanks for spotting ❤️

  and :option:`-X dev <-X>` is passed to the Python or Python is built in :ref:`debug mode <debug-build>`.
  (Contributed by Donghee Na in :gh:`141770`.)

> * The official Windows 64-bit binaries on python.org__ are now using the new

Reviewer 2: ```suggestion
* The official Windows 64-bit binaries on python.org__ now use the new
```

>   :ref:`tail-calling interpreter <whatsnew314-tail-call-interpreter>` since Python 3.15.0a8.

Reviewer 3: ```suggestion
  :ref:`tail-calling interpreter <whatsnew314-tail-call-interpreter>` since Python 3.15.
```

There's no need to call out the specific pre-release in whatsnew, right? That can be found in NEWS @hugovk

Reviewer 2: Yeah, and this page is all about 3.15, so no need to mention that:

```suggestion
  :ref:`tail-calling interpreter <whatsnew314-tail-call-interpreter>`.
```

>   (Contributed by Ken Jin, Brandt Bucher and Chris Eibl in :gh:`143068`.

Reviewer: I think technically you contributed this build change and not me nor the rest of us though?

Author: Yupp, I've done https://github.com/python/release-tools/pull/346 with @zooba's help. Don't think we can / do link to release-tools here?

Thus, I've re-used the wording of [#143068 blurb](https://github.com/python/cpython/pull/143068/changes#diff-029bba5f8a3aa614f4250be214ae90835069e7b54cf325b44b4e14d17eee6c28).

Maybe just link to https://docs.python.org/3.15/whatsnew/3.15.html#optimizations?

  and :option:`-X dev <-X>` is passed to the Python or Python is built in :ref:`debug mode <debug-build>`.
  (Contributed by Donghee Na in :gh:`141770`.)

.. _whatsnew315-windows-tail-calling-interpreter:

> * The official Windows 64-bit binaries on python.org__ now use the new

Reviewer: We should merge the other news entry in optimizations into here. Having two separate news entry is a little messy.
